Geometry_Engine/Query/CurvePlanarIntersections.cs

```diff
@@ -33,149 +34,85 @@ public static partial class Query
-
-            List<Point> iPts = circle1.CurvePlanarIntersections(curve2, tolerance);
-
-            Point midPoint1 = curve1.PointAtParameter(0.5);
-            double dist = midPoint1.Distance(curve1.StartPoint());
-
-            for (int i = iPts.Count - 1; i >= 0; i--)
-            {
-                if (midPoint1.Distance(iPts[i]) - dist > tolerance)
-                    iPts.RemoveAt(i);
-            }
-
-            return iPts;
+            return curve1.CurveIntersections(curve2, tolerance);
         }
 
         /***************************************************/
 
+        [DeprecatedAttribute("CurveIntersections")]
         public static List<Point> CurvePlanarIntersections(this Circle curve1, Arc curve2, double tolerance = Tolerance.Distance)
         {
-            return curve2.CurvePlanarIntersections(curve1, tolerance);
+            return curve1.CurveIntersections(curve2, tolerance);
         }
 
         /***************************************************/
 
+        [DeprecatedAttribute("CurveIntersections")]
         public static List<Point> CurvePlanarIntersections(this Circle curve1, Circle curve2, double tolerance = Tolerance.Distance)
         {
-            List<Point> iPts = new List<Point>();
-            Point c1 = curve1.Centre;
-            Point c2 = curve2.Centre;
-            double r1 = curve1.Radius;
-            double r2 = curve2.Radius;
-            Vector n1 = curve1.Normal;
-            Vector n2 = curve2.Normal;
-            Plane p1 = new Plane { Origin = c1, Normal = n1 };
-            Plane p2 = new Plane { Origin = c2, Normal = n2 };
-
-            if (!p1.IsCoplanar(p2))
-                return iPts;
-
-            double sqrDist = c1.SquareDistance(c2);
-            double dist = Math.Sqrt(sqrDist);
-
-            if (dist <= tolerance)
-                return iPts;
-
-            double sumRadii = r1 + r2;
-            double difRadii = Math.Abs(r1 - r2);
-            Vector dir = (c2 - c1).Normalise();
-
-            if (Math.Abs(dist - sumRadii) <= tolerance)
-                iPts.Add(c1 + dir * r1);
-            else if (Math.Abs(dist - difRadii) <= tolerance)
-                iPts.Add(c1 - dir * r1);
-            else if (dist - sumRadii < tolerance && dist - difRadii >= tolerance)
-            {
-                double a = (r1 * r1 - r2 * r2 + sqrDist) / (2 * dist);
-                Point midPt = (c1 + dir * a);
-                Vector perp = dir.Rotate(Math.PI * 0.5, n1);
-                double shift = Math.Sqrt(r1 * r1 - midPt.SquareDistance(c1));
-                iPts.Add(midPt + perp * shift);
-                iPts.Add(midPt - perp * shift);
-            }
-
-            return iPts;
+            return curve1.CurveIntersections(curve2, tolerance);
         }
 
         /***************************************************/
 
+        [DeprecatedAttribute("CurveIntersections")]
```

Comment:
Please check the format of the constructor for the deprecated attribute here.

Reply:
Please have a look if you're happy with changes.